cluster/log-dump/log-dump.sh

```diff
@@ -378,16 +394,16 @@ function dump_nodes() {
   local node_names=()
   local windows_node_names=()
   if [[ -n "${1:-}" ]]; then
-    echo "Dumping logs for nodes provided as args to dump_nodes() function"
+    echo 'Dumping logs for nodes provided as args to dump_nodes() function'
     node_names=( "$@" )
   elif [[ -n "${use_custom_instance_list}" ]]; then
-    echo "Dumping logs for nodes provided by log_dump_custom_get_instances() function"
-    node_names=( $(log_dump_custom_get_instances node) )
-  elif [[ ! "${node_ssh_supported_providers}" =~ "${KUBERNETES_PROVIDER}" ]]; then
+    echo 'Dumping logs for nodes provided by log_dump_custom_get_instances() function'
+    while IFS='' read -r line; do node_names+=("$line"); done < <(log_dump_custom_get_instances node)
+  elif [[ "${node_ssh_supported_providers}" != *"${KUBERNETES_PROVIDER}"* ]]; then
```

Comment:
It looks that it works the same, but what is the reason to use != *var* syntax instead of just skipping quotes?

Reply:
The reason:

https://github.com/koalaman/shellcheck/wiki/SC2076

Now, one alternative would of course be to leave the code as-is and add a `#shellcheck disable=SC2076` line above.
(IMO, it would be preferable not to disable just because we don't know why it is working/not working but to find out what is really going on and act accordingly)

Reply:
I was asking why don't you just drop quotes on the right-side?

Reply:
I find it less readable than solution from: https://github.com/kubernetes/kubernetes/pull/88349/commits/742701622abe36e4e8914db168c1ff926f479da6#diff-5d682968f2f3b4ff2f9f2bb9a303f650L92
does it work better?

Reply:
Let's try! I can upload a fix for this tomorrow morning.

Reply:
looks that it works fine :)